### PR TITLE
feat(2152): Support PRs for enterprise Sonar [1]

### PR DESCRIPTION
**Pull request by tkyi**
## Context

When a Pull Request Analysis runs, Screwdriver Sonar Coverage book end must make use of Pull Request Analysis.
Screwdriver should not create a new Sonar Project for each Pull Request. We should use [Pull Request Analysis](https://docs.sonarqube.org/latest/analysis/pull-request/) feature and use analysis data from Pull Request to populate coverage data on UI.

## Objective

This PR converts projectKey from `job:jobId` => `pipeline:pipelineId` if enterprise Sonar is in use. Also passes in PR number for pull request analysis.

## Related links
Related to https://github.com/screwdriver-cd/screwdriver/issues/2152
Analysis endpoint docs: https://sonar.screwdriver.ouroath.com/web_api/api/measures/search_history

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -213,8 +241,8 @@ class CoverageSonar extends CoverageBase {
      * @return {Promise}                   An access token that build can use to talk to coverage server
      */
     _getAccessToken(buildCredentials) {
-        const { jobId } = buildCredentials;
-        const projectKey = `job:${jobId}`;
+        const { jobId, pipelineId } = buildCredentials;
+        const projectKey = getProjectKey({ enterpriseEnabled: sonarEnterprise, jobId, pipelineId });
         const username = `user-job-${jobId}`;
```
Comment (jithine) [inline index.js:246]: We don't need separate users for each job, so this should also be `pipelineId`

File: commands.txt [hunk 2]
```diff
@@ -7,8 +7,10 @@ SONAR_TOKEN=`$SD_CURL_CMD_WRAPPER "curl -H \"Authorization: Bearer $SD_TOKEN\" $
 VERSION=`cat sonar-project.properties | awk -F= '/sonar.projectVersion=/{print $2}'`
 PROJECT_VERSION=${VERSION:='not provided'}
 SONAR_OPTIONS=`eval echo "$SD_SONAR_OPTS"`
+if [ "$SD_SONAR_ENTERPRISE" = false ]; then SD_SONAR_PROJECT_KEY=`job:$SD_JOB_ID`; else SD_SONAR_PROJECT_KEY=`pipeline:$SD_PIPELINE_ID`; fi
```
Comment (parthasl) [inline commands.txt:10]: for enterprise edition, we should allow jobId as project key as well, when overriding via env variable. will this be added in a new PR?

Comment (tkyi) [reply commands.txt:10]: yeah that would be second pass

File: index.js (hunk 1 above)
Comment (tkyi) [reply index.js]: updated

**Pull request merged**